preview-env/clean/action.yml

```diff
@@ -0,0 +1,444 @@
+          if [ -z $existing_comment_id ]; then
+            # Create a new comment
+            if [ "$DRY_RUN" = "true" ]; then
+              echo [DRY-RUN]: Creating a new comment ...
+            else
+              echo Creating a new comment ...
+              gh pr comment $pr_number --body "$new_comment_body"
+            fi
+          else
+            # Update existing comment
+            if [ "$DRY_RUN" = "true" ]; then
+              echo [DRY-RUN]: Updating existing comment ...
+            else
+              echo Updating existing comment ...
+              gh api \
+                --method PATCH \
+                /repos/$REPOSITORY_OWNER/$REPOSITORY_NAME/issues/comments/$existing_comment_id \
+                --field body="$new_comment_body" \
+                --silent
+            fi
+          fi
+        else
+          # Delete existing comment if any
+          if [ ! -z $existing_comment_id ]; then
+            if [ "$DRY_RUN" = "true" ]; then
+              echo [DRY-RUN]: Deleting existing comment ...
+            else
+              echo Deleting existing comment ...
+              gh api \
+                --method DELETE \
+                /repos/$REPOSITORY_OWNER/$REPOSITORY_NAME/issues/comments/$existing_comment_id \
+                --silent
+            fi
+          fi
+        fi
+      done
+    shell: bash
+  - name: Clean up Pull Request comments
+    env:
+      DRY_RUN: ${{ inputs.dry-run }}
+      GH_TOKEN: ${{ inputs.token || github.token }}
+      LABELS: ${{ inputs.labels }}
+      REPOSITORY_NAME: ${{ steps.inputs.outputs.repository-name }}
+      REPOSITORY_OWNER: ${{ steps.inputs.outputs.repository-owner }}
+    run: |
+      set -e
+
+      comment_ids_to_delete=""
+
+      # Get all open Pull Requests with their comments.
+      open_pull_requests=$(
+        gh api graphql \
+          --paginate \
+          --field owner=$REPOSITORY_OWNER \
+          --field repo=$REPOSITORY_NAME \
+          --field states=OPEN \
+          --field query=@${{ github.action_path }}/graphql/query_get_pull_requests_with_comments.gql \
+          --jq '.data.repository.pullRequests.nodes'
+      )
+
+      # Get the 100 most recent closed or merged Pull Requests with their comments
+      # There's no need to scan the entire history every time. Retrieving the last 100 closed or merged Pull Requests is enough to clean them up at least once.
+      closed_merged_pull_requests=$(
+        gh api graphql \
+          --field owner=$REPOSITORY_OWNER \
+          --field repo=$REPOSITORY_NAME \
+          --field 'states[]=CLOSED' \
+          --field 'states[]=MERGED' \
+          --field query=@${{ github.action_path }}/graphql/query_get_pull_requests_with_comments.gql \
+          --jq '.data.repository.pullRequests.nodes'
+      )
+
+      # If Pull Request is not OPEN, any warning or shutdown comments must be removed
+      # If Pull Request is OPEN and has no deploy labels, any warning comments must be removed
```

Comment:
maybe rephrase the comment

I assume it's supposed to delete the "delete in x days" comment but keep the "deleted env" comment?

Both include a "warning" in the message itself, which can be misleading.

Reply:
For the shutdown message, I suggest to replace the warning block with a note block to avoid confusion.
<img width="826" alt="image" src="https://github.com/camunda/infra-global-github-actions/assets/1970739/34b5a08d-64df-4fa7-b783-69ca41e45fd2">

> I assume it's supposed to delete the "delete in x days" comment but keep the "deleted env" comment?

Yes only for OPEN PRs with no preview labels attached.

Reply:
I got confused by it in the first place as this has `warning` as part of the message, so I thought it may delete it but doesn't since it's not the `warning` message that is meant.
![image](https://github.com/camunda/infra-global-github-actions/assets/9141483/be22e8bf-5a51-424e-8db8-b195a14ee1db)

Reply:
Let's switch to a note block 👍